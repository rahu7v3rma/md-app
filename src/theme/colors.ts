// theme colors
const PRIMARY = '#72996B';
const PRIMARY_LIGHT = '#D3E29F40';
const BACKGROUND = '#F5F5F5';
const SHEET_BACKGROUND_COLOR = 'rgba(0,0,0,0.5)';
const BACKGROUND_LIGHTEST = '#FAFAFF';
const LOG_PAGE_BACKGROUND_COLOR = '#f5f4ee';

// INPUT
const INPUT = '#D3E6F8';
const INPUT_BG = '#EEF4FA';
const INPUT_ERROR_BG = '#FFECEC';
const INPUT_ERROR_BORDER = '#F9D4D4';
const INPUT_ERROR_PLACEHOLDER = '#D05151';
const INPUT_BORDER = '#D3E29F';

// PROGRESS BAR
const PROGRESS_BAR_FILL = '#4DBA0B';
const PROGRESS_BAR_FILL_BORDER = '#3A9A00';
const PROGRESS_BAR_ACTIVITY_INDICATOR = PRIMARY;
const PROGRESS_BAR_FILL_GREEN = '#72996B';
const PROGRESS_BAR_FILL_DISABLE = '#BFC4C9';
const PROGRESS_BAR_TRACK_GREEN = '#EDF0DA';

// PROGRESS CIRCLE
const PROGRESS_CIRCLE_COLOR = '#f00';
const PROGRESS_CIRCLE_SHADOW_COLOR = '#999';
const PROGRESS_CIRCLE_BG_COLOR = '#e9e9ef';
const PROGRESS_CIRCLE_START_BALL_COLOR = '#72996B';
const PROGRESS_CIRCLE_START_BALL_DISABLE_COLOR = '#BDBECA';
const PROGRESS_CIRCLE_INNER_BALL_COLOR = '#5B37BE';
const PROGRESS_CIRCLE_INNER_BALL_DISABLE_COLOR = '#A4AAAF';

// SLIDER
const SLIDER_BG = 'rgba(120, 120, 128, 0.32)';

// BUTTON
const BUTTON_BORDER = '#D3E6F8';
const GREEN_CHECK_DARK = '#D3E6F8';
const BUTTON_LIGHTER_BORDER = '#f4f8e7';
const BUTTON_LIGHTER_BACKGROUND = '#f4f8e7';
const BUTTON_PRIMARY_BORDER = '#5A3DBF';
const BUTTON_DISABLED_BG = '#cccccc';
const BUTTON_DISABLED_BORDER = '#D6D6D6';
const BUTTON_IMAGE_BACKGROUND = '#F0E957';
const BUTTON_ICON_BACKGROUND = '#D3E29F40';
const BUTTON_RED_BACKGROUND = '#B16663';
const BUTTON_SUCCESS_BACKGROUND = '#4DBA0B';
const BUTTON_GREEN_BACKGROUND = '#72996B';
const BUTTON_WHITE_BACKGROUND = '#FFFFFF';
const BUTTON_DARK_RED_BACKGROUND = '#CD4049';

const TEXT_BLACK = '#271A51'; // more of a dark-blue
const TEXT_DARK_BLACK = '#1F1E22';
const TEXT_BLACK_GRAY = '#4D4F4D';
const TEXT_GRAY_BLACK = '#4D4F4D';
const TEXT_WHITE = '#FFFFFF';
const TEXT_GREEN = '#72996B';
const TEXT_CHARCOAL = '#4D4F4D';
const TEXT_DUSKY_OLIVE = '#847F60';
const TEXT_ERROR = '#D05151';
const TEXT_LIGHTER = 'rgba(235, 235, 245, 0.6)';
const TEXT_DISABLED = '#D6D6D6';
const TEXT_GRAY = '#A4AAAF';
const TEXT_GRAY_BASE = '#58595A';
const TEXT_PRIMARY = PRIMARY;
const TEXT_PURPLE = '#6E51D0';
const TEXT_LINK = '#3ED1FF';
const TEXT_DARK_GREEN = '#72996B';
const TEXT_DARKER = '#4D4F4D';
const TEXT_MAIN_DARKER = '#4D4F4D';

//ICONS
const EYE_ICON = '#6E51D0';
const EYE_ICON_GREEN = '#72996B';
const EYE_ICON_ERROR_BORDER = '#D05151';
const GREEN_CHECK_MARK = '#4DBA0B';

// toast messages
const ERROR_BACKGROUND = '#D05151';
const SUCCESS_BACKGROUND = '#4DBA0B';
const INFO_BACKGROUND = '#A4AAAF';

//Modals
const MODAL_BACKDROP = 'rgba(0,0,0,0.5)';
const MODAL_CLOSE_HANDLE = '#F5F8FB';
const MODAL_ITEM_DIVIDER = '#E6E8EC';

//EXTRAS
const SUCCESS_LIGHTEST = '#ECF8E6';
const SUCCESS_LIGHTER = '#DFEED8';
const DARK_CHARCOAL = '#1F1E22';
const WHITE = '#FFFFFF';
const BLACK = '#000000';
const BLACK_LIGHTER = 'rgba(37, 37, 37, 0.6)';
const JOURNEY_BACKGROUND = '#CEE1FE';
const JOURNEY_SECONDARY = '#ADCAF4';
const BLOCK_IN_PROGRESS = PRIMARY;
//const BLOCK_COMPLETE = '#4DBA0B';
const ERROR_LIGHTEST = '#FFECEC';
const ERROR_LIGHTER = '#F9D4D4';
const ERROR_BASE = '#D05151';
const BLACK_SHADOW = '#171717';
const CARD_COMPLETED = '#eff8e7';
const CARD_INPROGRESS = '#f0f4f8';
const PAGE_BG = '#F5F8FB';
const LIGHT_BACK_COLOR = '#D3E29F40';
const BLACK_CARDS = 'rgba(0, 0, 0, 0.3)';
const HEADER_SHADOW = 'rgba(0, 0, 0, 0.03)';
const GRAY_SHADOW = '#0000000D';
const BORDER_COLOR = '#E1E6C2';
const TRANSPARENT_COLOR = 'transparent';
const BADGE_RED = '#F50157';

//SWITCH
const SWITCH_TOGGLED = '#72996B';
const SWITCH_BUTTON = WHITE;

// pagination
const ACTIVE_PAGINATION_DOT = PRIMARY;
const INACTIVE_PAGINATION_DOT = PRIMARY_LIGHT;

export const Colors = {
    theme: {
        primary: PRIMARY,
        primary_light: PRIMARY_LIGHT,
        app_background: BACKGROUND,
        app_sheet_background_color: SHEET_BACKGROUND_COLOR,
        app_background_lightest: BACKGROUND_LIGHTEST,
        log_page_background_color: LOG_PAGE_BACKGROUND_COLOR
    },
    text: {
        black: TEXT_BLACK,
        black_gray: TEXT_BLACK_GRAY,
        dark_green: TEXT_DARK_GREEN,
        text_gray_black: TEXT_GRAY_BLACK,
        dark_black: TEXT_DARK_BLACK,
        white: TEXT_WHITE,
        error: TEXT_ERROR,
        green: TEXT_GREEN,
        charcoal: TEXT_CHARCOAL,
        dusky_olive: TEXT_DUSKY_OLIVE,
        text_lighter: TEXT_LIGHTER,
        text_disabled: TEXT_DISABLED,
        gray: TEXT_GRAY,
        gray_Base: TEXT_GRAY_BASE,
        primary: TEXT_PRIMARY,
        purple: TEXT_PURPLE,
        link: TEXT_LINK,
        darker: TEXT_DARKER,
        mainDarker: TEXT_MAIN_DARKER
    },
    input: {
        error_bg: INPUT_ERROR_BG,
        error_border: INPUT_ERROR_BORDER,
        app_input: INPUT,
        app_input_bg: INPUT_BG,
        input_placeholder_error: INPUT_ERROR_PLACEHOLDER,
        input_border: INPUT_BORDER,
        light_back_color: LIGHT_BACK_COLOR
    },
    button: {
        app_button_border: BUTTON_BORDER,
        app_button_lighter_background: BUTTON_LIGHTER_BACKGROUND,
        app_button_lighter_border: BUTTON_LIGHTER_BORDER,
        app_button_primary_border: BUTTON_PRIMARY_BORDER,
        app_button_disabled_bg: BUTTON_DISABLED_BG,
        app_button_disabled_border: BUTTON_DISABLED_BORDER,
        app_button_background: BUTTON_IMAGE_BACKGROUND,
        app_icon_button_background: BUTTON_ICON_BACKGROUND,
        app_button_red_background: BUTTON_RED_BACKGROUND,
        app_button_success_background: BUTTON_SUCCESS_BACKGROUND,
        app_button_green_background: BUTTON_GREEN_BACKGROUND,
        app_button_white_background: BUTTON_WHITE_BACKGROUND,
        app_btn_light_color: LIGHT_BACK_COLOR,
        app_button_dark_red_background: BUTTON_DARK_RED_BACKGROUND
    },
    icons: {
        eye: EYE_ICON,
        eyeGreen: EYE_ICON_GREEN,
        eye_error_border: EYE_ICON_ERROR_BORDER,
        green_check_mark: GREEN_CHECK_MARK,
        green_dark_check: GREEN_CHECK_DARK
    },
    modals: {
        backdrop: MODAL_BACKDROP,
        close_handle: MODAL_CLOSE_HANDLE,
        divider: MODAL_ITEM_DIVIDER
    },
    toast: {
        errorBackground: ERROR_BACKGROUND,
        successBackground: SUCCESS_BACKGROUND,
        infoBackground: INFO_BACKGROUND
    },
    progress: {
        fillWhite: TEXT_WHITE,
        fill: PROGRESS_BAR_FILL,
        fill_border: PROGRESS_BAR_FILL_BORDER,
        activity_indicator: PROGRESS_BAR_ACTIVITY_INDICATOR,
        fillGreen: PROGRESS_BAR_FILL_GREEN,
        fillDisable: PROGRESS_BAR_FILL_DISABLE,
        trackColor: PROGRESS_BAR_TRACK_GREEN,
        headerProgressColor: PROGRESS_CIRCLE_INNER_BALL_COLOR
    },
    slider: {
        bg: SLIDER_BG
    },
    extras: {
        success_lightest: SUCCESS_LIGHTEST,
        success_lighter: SUCCESS_LIGHTER,
        white: WHITE,
        black: BLACK,
        black_lighter: BLACK_LIGHTER,
        blue_background: JOURNEY_BACKGROUND,
        blue_secondary: JOURNEY_SECONDARY,
        block_in_progress: BLOCK_IN_PROGRESS,
        block_complete: BLOCK_IN_PROGRESS,
        error_lighter: ERROR_LIGHTER,
        error_lightest: ERROR_LIGHTEST,
        error_base: ERROR_BASE,
        black_shadow: BLACK_SHADOW,
        card_completed: CARD_COMPLETED,
        card_inProgress: CARD_INPROGRESS,
        page_bg: PAGE_BG,
        blackCards: BLACK_CARDS,
        headerShadow: HEADER_SHADOW,
        dark_charcoal: DARK_CHARCOAL,
        block_locked: INPUT_BORDER,
        gray_shadow: GRAY_SHADOW,
        border_color: BORDER_COLOR,
        transparent: TRANSPARENT_COLOR,
        badgeRed: BADGE_RED
    },
    pagination: {
        active_dot: ACTIVE_PAGINATION_DOT,
        inactive_dot: INACTIVE_PAGINATION_DOT
    },
    progressCircle: {
        color: PROGRESS_CIRCLE_COLOR,
        shadow_color: PROGRESS_CIRCLE_SHADOW_COLOR,
        bg_color: PROGRESS_CIRCLE_BG_COLOR,
        start_ball_color: PROGRESS_CIRCLE_START_BALL_COLOR,
        start_ball_disable_color: PROGRESS_CIRCLE_START_BALL_DISABLE_COLOR,
        inner_ball_color: PRIMARY,
        inner_ball_disable_color: PROGRESS_CIRCLE_INNER_BALL_DISABLE_COLOR
    },
    switch: {
        button: SWITCH_BUTTON,
        toggled: SWITCH_TOGGLED
    },
    statusBar: {
        white: WHITE
    }
};
