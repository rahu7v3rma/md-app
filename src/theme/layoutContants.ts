export const LayoutContants = {
    gapSpacing: 5,
    cardHeight: 90
};
