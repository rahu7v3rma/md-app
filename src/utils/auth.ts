import { CommonActions } from '@react-navigation/native';
import * as Keychain from 'react-native-keychain';

import { navigationRef } from '@/navigation';
import { resetNotifications } from '@/services/notification';
import { resetStore } from '@/store';

export const setAuthToken = (token: string) => {
    Keychain.setGenericPassword('token', token);
};

export const getAuthToken = (): Promise<string | null> => {
    return Keychain.getGenericPassword().then((credentials) =>
        credentials === false ? null : credentials.password
    );
};

export const AUTHORIZATION_HEADER_NAME = 'X-Authorization';

export const getAuthorizationHeaderValue = () => {
    return getAuthToken()
        .then((authToken) => `Token ${authToken || ''}`)
        .catch(() => undefined);
};

export const resetAuthToken = () => {
    Keychain.resetGenericPassword();
};

export const logoutAction = (skipNavigation: boolean = false) => {
    resetStore();
    resetNotifications();
    resetAuthToken();

    if (!skipNavigation) {
        navigationRef.dispatch(
            CommonActions.reset({
                index: 0,
                routes: [{ name: 'SignIn' }]
            })
        );
    }
};
