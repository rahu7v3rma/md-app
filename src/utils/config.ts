import config from 'react-native-config';

export const API_BASE_URL = config.API_BASE_URL;
