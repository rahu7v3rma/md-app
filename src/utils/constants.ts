import { ReactionData } from 'stream-chat-react-native';

import logActivity from '@/assets/svg/logActivity.svg';
import logBlood from '@/assets/svg/logBlood.svg';
import logFast from '@/assets/svg/logFastIcon.svg';
import logFood from '@/assets/svg/logFood.svg';
import logHydration from '@/assets/svg/logHydration.svg';
import logInsulin from '@/assets/svg/logInsulin.svg';
import logMedication from '@/assets/svg/logMedication.svg';
import logWeight from '@/assets/svg/logWeight.svg';
import {
    Heart,
    HighFive,
    Like,
    Muscle,
    SadFace,
    SmileyFace
} from '@/assets/svgs';

export const DateTimeFormat = 'DD/MM/YYYY HH:mm';

export const Constants: any = {
    trackerTitle: {
        tracked: 'Fast tracking is Paused',
        complete: 'Fast was completed',
        tracking: 'Fast in progress'
    },
    notificationType: {
        STREAM_CHAT_MESSAGE: 'message.new',
        LOG_REMINDER: 'log.reminder',
        LESSON_REMINDER: 'lesson.reminder',
        FAST_COMPLETE: 'fast.complete'
    },
    logs: {
        userFast: 'UserFast',
        userWeight: 'UserWeight',
        userInsulin: 'UserInsulin',
        userExercise: 'UserExercise',
        userGlucose: 'UserGlucose',
        userDrink: 'UserDrink',
        userMedication: 'UserMedication',
        userFood: 'UserFood',
        userLesson: 'UserLesson'
    },
    logTypeDetails: {
        UserGlucose: {
            icon: logBlood,
            title: 'Glucose',
            screen: 'LogBlood'
        },
        UserMedication: {
            icon: logMedication,
            title: 'Medication',
            screen: 'LogMedication'
        },
        UserDrink: {
            icon: logHydration,
            title: 'Drink',
            screen: 'LogWaterIntake'
        },
        UserWeight: {
            icon: logWeight,
            screen: 'LogWeight',
            title: 'Weight'
        },
        UserFast: {
            icon: logFast,
            title: 'Fast',
            screen: 'LogFast'
        },
        UserExercise: {
            icon: logActivity,
            title: 'Activity',
            screen: 'LogActivity'
        },
        UserInsulin: {
            icon: logInsulin,
            title: 'Insulin',
            screen: 'LogInsulin'
        },
        UserFood: {
            icon: logFood,
            title: 'Food',
            screen: 'LogMeal'
        },
        UserLesson: {
            icon: logActivity,
            title: 'Lesson Completed',
            screen: 'LessonContent'
        }
    },
    logTabsList: [
        {
            id: 1,
            icon: logBlood,
            title: 'Log Blood Glucose',
            screen: 'LogBlood',
            type: 'UserGlucose'
        },
        {
            id: 2,
            icon: logHydration,
            title: 'Log Hydration',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 3,
            icon: logActivity,
            title: 'Log Activity',
            screen: 'LogActivity',
            type: 'UserExercise'
        },
        {
            id: 4,
            icon: logFood,
            title: 'Log a Meal',
            screen: 'LogMeal',
            type: 'UserFood'
        },
        {
            id: 5,
            icon: logMedication,
            title: 'Log Medication',
            screen: 'LogMedication',
            type: 'UserMedication'
        },
        {
            id: 6,
            icon: logWeight,
            title: 'Log Weight',
            screen: 'LogWeight',
            type: 'UserWeight'
        },
        {
            id: 7,
            icon: logFast,
            title: 'Log a Fast',
            screen: 'LogFast',
            type: 'UserFast'
        },
        {
            id: 8,
            icon: logInsulin,
            title: 'Log Insulin',
            screen: 'LogInsulin',
            type: 'UserInsulin'
        }
    ],
    trackFast: {
        state: {
            tracking: 'tracking',
            tracked: 'tracked',
            complete: 'complete',
            empty: ''
        }
    },
    userDataList: [
        {
            name: 'Tamara Green',
            message: 'Hello! Did you mean this?',
            time: '14/03/2023 9:00'
        },
        {
            name: 'Adam Storm',
            message: 'Hello! Did you mean this?',
            time: '13/03/2023 08:20'
        },
        {
            name: 'Samantha Green',
            message: 'Hello! Did you mean this?',
            time: '12/03/2023 13:50'
        },
        {
            name: 'Kristen Stuart',
            message: 'Hello! Did you mean this?',
            time: '12/03/2023 10:05'
        },
        {
            name: 'Adam Storm',
            message: 'Hello! Did you mean this?',
            time: '05/03/2023 11:50'
        },
        {
            name: 'Tamara Green',
            message: 'Hello! Did you mean this?',
            time: '05/02/2023 09:50'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '06/02/2023 16:00'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '06/02/2023 15:00'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '06/02/2023 13:10'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '08/02/2023 17:50'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '01/01/2023 10:40'
        },
        {
            name: 'Samantha Green',
            message: 'Hello! Did you mean this?',
            time: '01/01/2023 21:30'
        },
        {
            name: 'Kristen Stuart',
            message: 'Hello! Did you mean this?',
            time: '02/01/2023 08:10'
        },
        {
            name: 'Adam Storm',
            message: 'Hello! Did you mean this?',
            time: '02/01/2023 12:00'
        },
        {
            name: 'Tamara Green',
            message: 'Hello! Did you mean this?',
            time: '03/01/2023 16:00'
        },
        {
            name: 'Samantha Green',
            message: 'Hello! Did you mean this?',
            time: '06/01/2023 19:00'
        },
        {
            name: 'Kristen Stuart',
            message: 'Hello! Did you mean this?',
            time: '08/01/2023 11:40'
        }
    ],
    confirmationDialog: {
        title: {
            discard: 'Do you want to discard your changes?',
            delete: 'Are you sure you want to delete this log?'
        }
    },
    errors: {
        futureTime: 'You can not select future time.',
        fastDuration: 'The logged time should be greater than 0',
        exerciseDuration: 'The exercise duration should be greater than 0',
        logValue: 'The log value should be greater than 0'
    },
    maxFontSizeMultiplier: 1.2
};

export const customReactionData: ReactionData[] = [
    {
        Icon: Like,
        type: '+1'
    },
    {
        Icon: Heart,
        type: 'heart'
    },
    {
        Icon: Muscle,
        type: 'muscle'
    },
    // {
    //     Icon: SmileyFace,
    //     type: 'slightly_smiling_face'
    // },
    {
        Icon: SadFace,
        type: 'cry'
    },
    {
        Icon: HighFive,
        type: 'raised_hands'
    }
];
