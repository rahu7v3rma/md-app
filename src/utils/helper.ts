export const padNum = (num: number) => {
    return (num < 10 ? '0' : '') + num;
};

export const isTimerValid = (timer: number) => {
    return timer > 0;
};

export const isUnitValid = (unit: number) => {
    return unit > 0;
};

// Format number to restrict specified digits max after decimal point
export const formatNumber = (num: number, digits = 2) => {
    return +num.toFixed(digits);
};
