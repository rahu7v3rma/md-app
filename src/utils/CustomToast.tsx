import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import Toast, { ToastConfigParams } from 'react-native-toast-message';

import { CheckIcon, CrossIcon } from '@/assets/svgs';
import { Text } from '@/shared';
import { Size } from '@/shared/text';
import { Colors } from '@/theme/colors';

const toastConfig = {
    successResponse: ({ text1 }: ToastConfigParams<any>) => (
        <View style={[styles.container, styles.successContainer]}>
            <View style={styles.errorIconStyle}>
                <CheckIcon width={14} height={14} color={Colors.text.black} />
            </View>
            <Text
                style={styles.titleStyle}
                size={Size.XXSmall}
                color={Colors.text.white}
                fontWeight="600"
            >
                {text1}
            </Text>
        </View>
    ),
    errorResponse: ({ text1 }: ToastConfigParams<any>) => (
        <View style={[styles.container, styles.errorContainer]}>
            <TouchableOpacity
                style={styles.errorIconStyle}
                onPress={() => Toast.hide()}
            >
                <CrossIcon width={12} height={12} color={Colors.text.error} />
            </TouchableOpacity>
            <Text
                style={styles.titleStyle}
                size={Size.XXSmall}
                color={Colors.text.white}
            >
                {text1}
            </Text>
        </View>
    ),
    info: ({ text1 }: ToastConfigParams<any>) => (
        <View style={[styles.container, styles.infoContainer]}>
            <Text
                style={styles.titleStyle}
                size={Size.XXSmall}
                color={Colors.text.white}
            >
                {text1}
            </Text>
        </View>
    )
};

export default toastConfig;

const styles = StyleSheet.create({
    titleStyle: {
        marginLeft: 15
    },
    successIconStyle: {
        width: 25,
        height: 25,
        borderRadius: 25,
        backgroundColor: Colors.extras.white,
        justifyContent: 'center',
        alignItems: 'center'
    },
    errorIconStyle: {
        width: 17,
        height: 17,
        borderRadius: 17,
        backgroundColor: Colors.extras.white,
        justifyContent: 'center',
        alignItems: 'center'
    },
    container: {
        height: 55,
        width: '90%',
        borderRadius: 20,
        flexDirection: 'row',
        alignItems: 'center',
        paddingHorizontal: 16,
        marginBottom: 65
    },
    successContainer: {
        backgroundColor: Colors.toast.successBackground
    },
    errorContainer: {
        backgroundColor: Colors.toast.errorBackground
    },
    infoContainer: {
        backgroundColor: Colors.toast.infoBackground
    }
});
