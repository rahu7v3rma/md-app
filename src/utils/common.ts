import { once } from 'lodash';
import { PixelRatio, Platform } from 'react-native';

import { API_BASE_URL } from './config';

const warnLocalBackend = once(() =>
    console.warn('Working with a local backend')
);

export const COMMON = {
    get isIos() {
        return Platform.OS === 'ios';
    },
    get apiBaseUrl() {
        if (API_BASE_URL) {
            return API_BASE_URL;
        } else {
            warnLocalBackend();

            return this.isIos
                ? 'http://127.0.0.1:8000/'
                : 'http://10.0.2.2:8000/';
        }
    },
    responsiveSize(value: number) {
        return PixelRatio.getPixelSizeForLayoutSize(value) * 0.35;
    },
    stringFormat(s: string, ...args: any[]) {
        return s.replace(/{([0-9]+)}/g, (match, index) =>
            typeof args[index] === 'undefined' ? match : args[index]
        );
    },
    delay(ms: number) {
        return new Promise((resolve) => setTimeout(resolve, ms));
    },
    retryOperation(
        operation: () => Promise<any>,
        delayMs: number,
        retries: number
    ) {
        return new Promise<any>((resolve, reject) => {
            return operation()
                .then(resolve)
                .catch((reason) => {
                    if (retries > 0) {
                        return COMMON.delay(delayMs)
                            .then(
                                COMMON.retryOperation.bind(
                                    null,
                                    operation,
                                    delayMs,
                                    retries - 1
                                )
                            )
                            .then(resolve)
                            .catch(reject);
                    }

                    return reject(reason);
                });
        });
    }
};
