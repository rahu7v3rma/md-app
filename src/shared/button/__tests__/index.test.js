import React from 'react';
import { StyleSheet } from 'react-native';
import renderer, { act } from 'react-test-renderer';

import Button from '@/shared/button';
import { Colors } from '@/theme/colors';
import { COMMON } from '@/utils/common';

const buttonStyle = StyleSheet.create({
    wrapper: {
        justifyContent: 'center',
        alignItems: 'center',
        paddingHorizontal: 15.5,
        paddingVertical: COMMON.isIos ? 24 : 17.5,
        borderRadius: 20,
        margin: 0
    },
    primary: {
        backgroundColor: Colors.theme.primary,
        borderColor: Colors.theme.primary
    }
});

it('button snapshot', async () => {
    const tree = renderer.create(
        <Button
            primary
            block
            bordered={false}
            style={{}}
            disabled={false}
            onPress={() => {}}
        />
    );
    expect(tree).toMatchSnapshot();
});

it('should trigger the onPress event when clicked', async () => {
    const onPress = jest.fn();
    const tree = renderer.create(
        <Button
            primary
            block
            bordered={false}
            style={{}}
            disabled={false}
            onPress={onPress}
            testID="button"
        />
    );
    const touchArea = tree.root.findByProps({ activeOpacity: 0.75 }).props;
    await act(() => touchArea.onPress());
    expect(onPress).toHaveBeenCalledTimes(1);
});

it('should be disabled when the disabled prop is true', async () => {
    const tree = renderer.create(
        <Button
            bordered={false}
            style={{}}
            disabled={true}
            onPress={() => {}}
            testID="button"
        />
    );
    const button = tree.root.findByProps({ testID: 'button' }).props;
    expect(button.disabled).toBe(true);
    const touchArea = tree.root.findByProps({ activeOpacity: 1 }).props;
    expect(touchArea.disabled).toBe(true);
});

it('should have the correct styles for the primary variant', async () => {
    const tree = renderer.create(
        <Button
            primary
            bordered={false}
            style={{}}
            disabled={false}
            onPress={() => {}}
            testID="button"
        />
    );
    const touchArea = tree.root.findByProps({ activeOpacity: 0.75 }).props;
    expect(touchArea.style).toStrictEqual([
        buttonStyle.wrapper,
        false,
        buttonStyle.primary,
        false,
        undefined,
        undefined,
        {}
    ]);
});
