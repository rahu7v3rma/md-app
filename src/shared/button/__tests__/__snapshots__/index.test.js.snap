// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#72996B",
      "borderColor": "#72996B",
      "borderRadius": 20,
      "justifyContent": "center",
      "margin": 0,
      "opacity": 1,
      "paddingHorizontal": 15.5,
      "paddingVertical": 24,
      "width": "100%",
    }
  }
/>
`;
